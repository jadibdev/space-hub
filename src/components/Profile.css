table {
  width: 31.25rem;
  align-items: center;
}

table,
tr,
td {
  border: 1px solid #000;
}
