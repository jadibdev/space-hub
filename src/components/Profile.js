/* eslint-disable consistent-return */
/* eslint-disable array-callback-return */
import React from 'react';
import { useSelector } from 'react-redux';
import './Profile.css';

/*
  PROFILE SHOULD SHOW ONLY RESERVED MISSIONS
*/

const Profile = () => {
  const { missions, joinedMissions } = useSelector((state) => state.missions);
  const rock = useSelector((state) => state.rockets);
  const reservedRocket = rock.filter((book) => book.reserved === true);
  return (
    <div>
      <div>
        {missions.map((mission) => {
          if (joinedMissions.includes(mission.mission_id)) {
            return (
              <div key={mission.mission_id}>{mission.mission_name}</div>
            );
          }
        })}
        { joinedMissions.length === 0 ? 'Currently no missions!' : 'Enjoy your mission!'}
      </div>
      <div>
        <h1>MY Rockets</h1>
        <table>
          <tr>
            <td>
              {
            reservedRocket.map((rocket) => {
              const { name } = rocket;
              return (
                <>
                  <h3>{name}</h3>
                </>
              );
            })
            }
            </td>
          </tr>
        </table>
      </div>
    </div>
  );
};

export default Profile;
