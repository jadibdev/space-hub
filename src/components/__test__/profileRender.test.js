import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Profile from '../Missions';

it('Profile component renders successfully!', () => {
  const tree = renderer
    .create(<Provider store={store}><Profile /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
