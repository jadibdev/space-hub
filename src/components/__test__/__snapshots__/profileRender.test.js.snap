// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders successfully! 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Item name
        </th>
        <th>
          Item description
        </th>
        <th>
          Status
        </th>
        <th>
          -
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
